addons/auth_signup/models/res_users.py

```diff
@@ -183,15 +184,10 @@ def _action_reset_password(self):
         self.mapped('partner_id').signup_prepare(signup_type="reset", expiration=expiration)
 
         # send email to users with their signup url
-        template = False
+        account_created_template = None
         if create_mode:
-            try:
-                template = self.env.ref('auth_signup.set_password_email', raise_if_not_found=False)
-            except ValueError:
-                pass
-        if not template:
-            template = self.env.ref('auth_signup.reset_password_email')
-        assert template._name == 'mail.template'
+            account_created_template = self.env.ref('auth_signup.set_password_email', raise_if_not_found=False)
+            assert not account_created_template or account_created_template._name == 'mail.template'
```

Comment:
Do we want to crash in that case ? Shouldn't we log and continue ?

Reply:
we can